chainerx_cc/chainerx/native/native_device/linalg.cc

```diff
@@ -0,0 +1,156 @@
+#include "chainerx/native/native_device.h"
+
+#include <cmath>
+#include <cstdint>
+#include <type_traits>
+
+#include "chainerx/array.h"
+#include "chainerx/backend.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/indexable_array.h"
+#include "chainerx/kernels/creation.h"
+#include "chainerx/kernels/linalg.h"
+#include "chainerx/macro.h"
+#include "chainerx/native/data_type.h"
+#include "chainerx/native/elementwise.h"
+#include "chainerx/native/kernel_regist.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+// gesv
+extern "C" void dgesv_(int* n, int* nrhs, double* a, int* lda, int* ipiv, double* b, int* ldb, int* info);
+extern "C" void sgesv_(int* n, int* nrhs, float* a, int* lda, int* ipiv, float* b, int* ldb, int* info);
+
+// getrf
+extern "C" void dgetrf_(int* m, int* n, double* a, int* lda, int* ipiv, int* info);
+extern "C" void sgetrf_(int* m, int* n, float* a, int* lda, int* ipiv, int* info);
+
+// getri
+extern "C" void dgetri_(int* n, double* a, int* lda, int* ipiv, double* work, int* lwork, int* info);
+extern "C" void sgetri_(int* n, float* a, int* lda, int* ipiv, float* work, int* lwork, int* info);
```

Comment:
Let's guard these declarations too with `CHAINERX_ENABLE_LAPACK`.

Reply:
Then the places where they are used should be also covered with `CHAINERX_ENABLE_LAPACK`, it won't compile otherwise?

Reply:
I think changing the body to the following is fine. You could also create a free function and call it in `A.`.
```c++
#ifdef CHAINERX_ENABLE_LAPACK
    // A. Implementation using LAPACK.
#else 
    // B. throw...
#endif
```

Reply:
Done.